server/service/src/invoice_line/outbound_shipment_line/update/generate.rs

```diff
@@ -123,11 +123,17 @@ fn generate_line(
     if let Some(number_of_packs) = input.number_of_packs {
         update_line.number_of_packs = u32_to_i32(number_of_packs);
     }
-    if let Some(total_before_tax) = input.total_before_tax {
-        update_line.total_before_tax = total_before_tax;
+
+    update_line.total_before_tax = if let Some(total_before_tax) = input.total_before_tax {
+        total_before_tax
+    } else if let Some(number_of_packs) = input.number_of_packs {
+        update_line.sell_price_per_pack * number_of_packs as f64
+    } else if input.stock_line_id.is_some() || input.item_id.is_some() {
+        sell_price_per_pack * number_of_packs as f64
     } else {
-        update_line.total_before_tax = sell_price_per_pack * number_of_packs as f64;
-    }
+        sell_price_per_pack * number_of_packs as f64
+    };
```

Comment:
The two if else and the last else block all do the same, right? maybe just use the last else?

Reply:
🤔  That's what I thought... [but was told to add it in](https://github.com/openmsupply/open-msupply/pull/306#pullrequestreview-1042585516)

Reply:
I might be wrong but think @andreievg just wanted to be very clear about the logic but not how exactly it is implemented (?). Find the ifs in combination with the last else confusing, maybe put it in a comment when the total after tax should be recalculated and why do the last else.

Reply:
The last else was supposed to be total before tax 🤔

Reply:
The logic looks sound, but yeah a comment would be great

Reply:
Ohh sorry, it looks sounds now, i just saw a commit you did